### PR TITLE
Add category filter to display books with respect to current selected filter

### PR DESCRIPTION
**Pull request by adejam**
The PR does the following:

- Add `filter` to the redux store which keeps the information about the current books filter.
- Add the `CategoryFilter` component to display a select box to select a filter option.
- Implement a method that modifies the `filter` state in the Redux store.
- Implement a conditional method that decides which books should be shown in the table depending on the `filter` value from the Redux store.
- [Live demo](https://deploy-preview-4--booklyb.netlify.app/)

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  const filteredBooks = () => {
+    let newBooks;
+    if (filterState === 'All') {
+      newBooks = books;
+    } else {
+      newBooks = books.filter(book => book.category === filterState);
+    }
+    return newBooks;
```
Comment (bolah2009) [inline src/components/BooksList.js:22]: Why not return immedatlly in the first if then there won't be a need for the else case. You can return the filtered book immedialty there too. Makes the code more straightforward without unnecessary variables.

File: src/components/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import CategoryFilter from '../containers/CategoryFilter';
+
+const BooksList = ({ changeFilter }) => {
+  const books = useSelector(state => state.books);
+  const filterState = useSelector(state => state.filter);
+  const handleFilterChange = e => {
+    changeFilter(e.target.value);
+  };
+
+  const filteredBooks = () => {
```
Comment (bolah2009) [inline src/components/BooksList.js:15]: Filtering should be a redux action, no? 🤔 
Why is the logic here?

**Review by bolah2009 — APPROVED** [review]
Looks good to me! Keep up the good job!

File: src/components/BooksList.js (hunk 2 above)
Comment (adejam) [reply src/components/BooksList.js:15]: It's part of the project requirement to change the `filter` state with redux. Using redux to also filter would make changing the `filter` state redundant

**Pull request merged**